File: sdk/php/src/Resources/ProxmoxBackup.php
<?php

namespace HiTechCloud\SDK\Resources;

class ProxmoxBackup extends BaseResource
{
    /**
     * Get connection info
     */
    public function getConnectionInfo(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/pbs');
    }

    /**
     * Get credentials
     */
    public function getCredentials(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/pbs/credentials');
    }

    /**
     * Get usage
     */
    public function getUsage(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/pbs/usage');
    }

    /**
     * Get metrics
     */
    public function getMetrics(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/pbs/metrics');
    }

    /**
     * List snapshots
     */
    public function listSnapshots(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/pbs/snapshots');
    }

    /**
     * List groups
     */
    public function listGroups(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/pbs/groups');
    }

    /**
     * Change password
     */
    public function changePassword(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/pbs/password', $data);
    }

    /**
     * Rotate token
     */
    public function rotateToken(int $serviceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/pbs/token');
    }

    /**
     * Revoke token
     */
    public function revokeToken(int $serviceId): array
    {
        return $this->http->delete('/api/service/' . $serviceId . '/pbs/token');
    }

}

File: sdk/csharp/Resources/ProxmoxBackup.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class ProxmoxBackup : BaseResource
{
    public ProxmoxBackup(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> PbsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/pbs", id));
    }

    public async Task<Dictionary<string, object?>> CredentialsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/pbs/credentials", id));
    }

    public async Task<Dictionary<string, object?>> UsageAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/pbs/usage", id));
    }

    public async Task<Dictionary<string, object?>> MetricsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/pbs/metrics", id));
    }

    public async Task<Dictionary<string, object?>> SnapshotsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/pbs/snapshots", id));
    }

    public async Task<Dictionary<string, object?>> GroupsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/pbs/groups", id));
    }

    public async Task<Dictionary<string, object?>> ChangePasswordAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/pbs/password", id), data);
    }

    public async Task<Dictionary<string, object?>> CreateTokenAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/pbs/token", id), data);
    }

    public async Task<Dictionary<string, object?>> DeleteTokenAsync(int id)
    {
        return await Http.DeleteAsync(string.Format("/api/service/{id}/pbs/token", id));
    }
}
